genestack_client/genestack_connection.py

```diff
@@ -239,8 +239,11 @@ def perform_request(self, path, data='', follow=True, headers=None):
         if headers:
             _headers.update(headers)
         try:
-            response = self.session.post(self.server_url + path, data=data, headers=_headers,
-                                         allow_redirects=follow)
+            response = self.session.post(
+                self.server_url + path, data=data, headers=_headers,
+                allow_redirects=follow,
+                timeout=int(os.environ.get("GENESTACK_CLIENT_TIMEOUT"))
+                if os.environ.get("GENESTACK_CLIENT_TIMEOUT", None) else None)
```

Comment:
It's unreadable, badly indented and crashes in case of non-numerical value.
Timeout value should be retrieved from env once, then converted to `int` (or even `float`) with a safeguard against invalid values.